src/ui/Tab/ResponsiveTabs.ts

```diff
@@ -0,0 +1,178 @@
+      let tabSectionElement = <HTMLElement>document.querySelector(this.coveoTabSection);
+
+      if (!tabSectionElement) {
+        return;
+      }
+      this.tabSection = $$(tabSectionElement);
+
+      let manageResponsiveTabs = () => {
+        this.toggleSmallTabsIfNeeded();
+        
+        let tabSectionIsOverflowing = this.isOverflowing(this.tabSection.el);
+        if (tabSectionIsOverflowing && this.tabSection.is('.coveo-small-tab-section')) {
+          let currentTab;
+          let overflowingElements = [];
+          let tabs = this.tabSection.findAll('.CoveoTab');
+          
+          this.tabSection.el.appendChild(this.dropdownHeader.el);
+          for(let i = tabs.length - 1; i >= 0; i--) {
+            currentTab = tabs[i];
+            this.addToDropdown(currentTab);
+            if(!this.isOverflowing(this.tabSection.el)) {
+              break;
+            }
+          }
+          
+        } else if(!tabSectionIsOverflowing && this.tabSection.is('.coveo-small-tab-section') && !this.isDropdownEmpty()) {
+          let dropdownTabs = this.dropdownContent.findAll('.coveo-small-tab');
+          
+          while (!this.isOverflowing(this.tabSection.el) && !this.isDropdownEmpty()) {
+            let current = dropdownTabs.shift();
+            this.removeFromDropdown(current);
+            $$(current).insertBefore(this.dropdownHeader.el);
+          }
+          
+          if (this.isOverflowing(this.tabSection.el)) {
+            let tabs = this.tabSection.findAll('.coveo-small-tab');
+            this.addToDropdown(tabs.pop());
+          }
+          
+          if (this.isDropdownEmpty()) {
+            this.dropdownHeader.detach();
+            this.toggleSmallTabsIfNeeded();
+          }
+        }
+        
+      };
+      manageResponsiveTabs();
+      window.addEventListener('resize', <EventListener>_.debounce(manageResponsiveTabs, 200));
+    });
+  }
+
+
+  private static toggleSmallTabsIfNeeded() {
+    let tabSectionIsOverflowing = this.isOverflowing(this.tabSection.el);
+    if (tabSectionIsOverflowing && !this.tabSection.is('.coveo-small-tab-section')) {
+          this.toggleSmallClass(this.tabSection);
+          
+          if (this.searchBarElement) {
+            this.tabSection.insertAfter(this.searchBarElement);
+          }
+    } else if (this.tabSection.is('.coveo-small-tab-section')
+                   && !this.isLargeFormatOverflowing(this.tabSection.el)
+                   && this.isDropdownEmpty()) {
+      this.toggleSmallClass(this.tabSection);
+      
+      if (this.searchBarElement) {
+        this.tabSection.insertBefore(this.searchBarElement);
+      }
+    }
+  }
+  
+  private static isLargeFormatOverflowing(tabSectionElement: HTMLElement){
+    let virtualTabSection = $$(<HTMLElement>tabSectionElement.cloneNode(true));
+    ResponsiveTabs.toggleSmallClass(virtualTabSection);
+    
+    virtualTabSection.el.style.position = "absolute";
```

Comment:
single quote everywhere (you should run the ts lint / gulp prettify task on your code)